src/test/java/com/cloudant/fdblucene/FDBIndexInputTest.java

```diff
@@ -0,0 +1,71 @@
+package com.cloudant.fdblucene;
+
+import static org.junit.Assert.assertEquals;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.Random;
+import org.apache.lucene.store.IndexInput;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class FDBIndexInputTest extends BaseFDBTest {
+
+  @Parameters
+  public static Collection<Object[]> data() {
+    final Random random = new Random();
+    final Collection<Object[]> result = new LinkedList<Object[]>();
+    for (int i = 0; i < 1000; i++) {
+      final int pageSize = 10 + random.nextInt(5000);
+      final int pageCount = 2 + random.nextInt(5);
+      
+      final int len = random.nextInt(pageSize * pageCount);
+      final int off = random.nextInt((pageSize * pageCount) - len);
+
+      result.add(new Object[] {off, len, pageCount, pageSize});
+    }
+    return result;
+  }
+
+  private final long off;
+  private final long length;
+  private final int pageCount;
+  private final int pageSize;
+
+  public FDBIndexInputTest(long off, long length, int pageCount, int pageSize) {
```

Comment:
Use `final` for method args where possible.

Reply:
done